pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -103,48 +115,193 @@ type controllerVolumeTopology struct {
 	nodeMgr node.Manager
 }
 
+// wcpControllerVolumeTopology implements the commoncotypes.ControllerTopologyService
+// interface for workload flavor. It stores the necessary kubernetes configurations and informers required
+// to implement the methods in the interface.
+type wcpControllerVolumeTopology struct {
+	//k8sConfig is the in-cluster config for client to talk to the api-server.
+	k8sConfig *restclient.Config
+	// azInformer is an informer instance on the AvailabilityZone custom resource.
+	azInformer cache.SharedIndexInformer
+}
+
 // InitTopologyServiceInController returns a singleton implementation of the
 // commoncotypes.ControllerTopologyService interface.
 func (c *K8sOrchestrator) InitTopologyServiceInController(ctx context.Context) (
```

Comment:
I see a lot of code changed in `InitTopologyServiceInController` without feature-gate check.

to keep this clean, you can create two method `InitTopologyServiceInController` for vanilla and `InitTopologyServiceInWCPController`. 

this will also help protect code change under feature-gate.

Reply:
Talked to Divyen about this. We agreed to let it be.